libparsec/crates/crypto/src/rustcrypto/private.rs

```diff
@@ -134,13 +134,11 @@ impl PrivateKey {
     }
 
     pub fn generate_shared_secret_key(&self, peer_public_key: &PublicKey) -> SecretKey {
-        SecretKey::from(x25519(*self.0.as_bytes(), *peer_public_key.0.as_bytes()))
+        SecretKey::from(x25519(self.0.to_bytes(), peer_public_key.0.to_bytes()))
```

Comment:
why this change ?
`to` is supposed to be slow instead of `as`

Reply:
And you added the method for `self`, so you don't need to call `.0`

Reply:
Read the commit description